csrc/device_lower/pass/alias_memory.cpp

```diff
@@ -1925,16 +1925,50 @@ class PromoteReuseSyncModifier : private kir::ExprMutator {
   using kir::ExprMutator::dispatch;
 
   void dispatch(Expr* expr) final {
-    auto position = allocation_info_map_.getScopeMap().getExprPos(expr);
+    int prev_position = position_;
+    position_ = allocation_info_map_.getScopeMap().getExprPos(expr);
+
+    if (position_ - prev_position > 1) {
+      for (auto [start, end] : sync_intervals_) {
+        if (start >= prev_position && start < position_ - 1) {
```

Comment:
Trying to remember the overall logic here, but shouldn't this be:

```
if (start >= prev_position && end< position_ - 1) {
```

I'm assuming this condition corresponds to:
> If a sync interval is contained entirely within the skipped expression range

Reply:
Sorry it's not very clear. I will update the comment.

This condition is just for whether we have skipped the beginning of the sync interval; this block is meant to ensure they are added to `upcoming_first_writes_`. In the next block, we cover cases where we skipped the end of the interval, whether or not we also skipped the beginning, and we translate those so that they are caught in the subsequent (existing) checks.